examples/azure-openai-service/src/main.rs

```diff
@@ -0,0 +1,105 @@
+use std::error::Error;
+
+use async_openai::{
+    config::AzureConfig,
+    types::{
+        ChatCompletionRequestMessageArgs, CreateChatCompletionRequestArgs,
+        CreateCompletionRequestArgs, CreateEmbeddingRequestArgs, Role,
+    },
+    Client,
+};
+use futures::StreamExt;
+
+async fn chat_completion_example(client: &Client<AzureConfig>) -> Result<(), Box<dyn Error>> {
+    let request = CreateChatCompletionRequestArgs::default()
+        .max_tokens(512u16)
+        .model("gpt-3.5-turbo")
+        .messages([
+            ChatCompletionRequestMessageArgs::default()
+                .role(Role::System)
+                .content("You are a helpful assistant.")
+                .build()?,
+            ChatCompletionRequestMessageArgs::default()
+                .role(Role::User)
+                .content("How does large language model work?")
+                .build()?,
+        ])
+        .build()?;
+
+    let response = client.chat().create(request).await?;
+
+    println!("\nResponse:\n");
+    for choice in response.choices {
+        println!(
+            "{}: Role: {}  Content: {}",
+            choice.index, choice.message.role, choice.message.content
+        );
+    }
+    Ok(())
+}
+
+async fn completions_stream_example(client: &Client<AzureConfig>) -> Result<(), Box<dyn Error>> {
+    let request = CreateCompletionRequestArgs::default()
+        .model("text-davinci-003")
+        .n(1)
+        .prompt("Tell me a short bedtime story about Optimus Prime and Bumblebee in Sir David Attenborough voice")
+        .stream(true)
+        .max_tokens(512_u16)
+        .build()?;
+
+    let mut stream = client.completions().create_stream(request).await?;
+
+    while let Some(response) = stream.next().await {
+        match response {
+            Ok(ccr) => ccr.choices.iter().for_each(|c| {
+                print!("{}", c.text);
+            }),
+            Err(e) => eprintln!("{}", e),
+        }
+    }
+    Ok(())
+}
+
+async fn embedding_example(client: &Client<AzureConfig>) -> Result<(), Box<dyn Error>> {
+    let request = CreateEmbeddingRequestArgs::default()
+        .model("text-embedding-ada-002")
+        .input([
+            "Why do programmers hate nature? It has too many bugs.",
+            "Why was the computer cold? It left its Windows open.",
```

Comment:
btw I think azure only supports one input at the time. I'm happy to test this out if needed. https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#embeddings

Reply:
Ah yes, this would be easy fix identified in previous dicussion.